src/telit_he910.cpp

```diff
@@ -0,0 +1,2102 @@
+
+}
+
+void openxc::telitHE910::commandCheck(TelitDevice* device) {
+
+	static unsigned int state = 0;
+	static bool first = true;
+	static unsigned long timer = 0xFFFF;
+
+	switch(state)
+	{
+		default:
+			state = 0;
+		case 0:
+			// check interval if it's not our first time
+			if(!first)
+			{
+				// request commands on interval
+				if(uptimeMs() - timer > GET_COMMANDS_INTERVAL)
+				{
+					timer = uptimeMs();
+					state = 1;
+				}
+			}
+			else
+			{
+				first = false;
+				timer = uptimeMs();
+				state = 1;
+			}
+			break;
+			
+		case 1:
+			if(!isSocketOpen(GET_COMMANDS_SOCKET))
+			{
+				if(!openSocket(GET_COMMANDS_SOCKET, device->config.serverConnectSettings))
+				{
+					state = 0;
+				}
+				else
+				{
+					state = 2;
+				}
+			}
+			else
+			{
+				state = 2;
+			}
+			break;
+			
+		case 2:
+			// call the GETconfigure API
+			switch(serverGETcommands(device->deviceId, device->config.serverConnectSettings.host))
+			{
+				case None:
+				case Working:
+					// if we are working, do nothing
+					break;
+				default:
+				case Success:
+					// send the contents of commandBuffer to the command handler
+					commands::handleIncomingMessage(commandBuffer, pCommandBuffer - commandBuffer, &(telitDevice->descriptor));
+					pCommandBuffer = commandBuffer;
+					state = 0;
+					break;
+				case Failed:
+					pCommandBuffer = commandBuffer;
+					closeSocket(GET_COMMANDS_SOCKET);
+					state = 0;
+					break;
+			}
+			break;
+	}
+
+	return;
+
+}
+
+/*PIPELINE*/
+
+/*
+ * Public:
+ *
+ * During pipeline::process(), this function will be called. The TelitDevice contains a QUEUE 
+ * that already has unprocessed data from previous calls to pipeline::publish. Here we simply need
+ * to flush the send QUEUE through a TCP/IP socket by constructing an HTTP POST. We can do this on 
+  * every call, or only when the QUEUE reaches a certain size and/or age.
+ */
+void openxc::telitHE910::processSendQueue(TelitDevice* device) {
+
+	// The pipeline is made up of a limited 256 byte queue_uint8_t defined in bytebuffer and emqueue
+	// The job of a processSendQueue device method is to empty the pipeline
+	// Normally we would just dump that queue straight out of the device (UART, USB)
+	// Right now, we are reformatting the pipeline data into a JSON array and POSTing via httpClient
+	// However, it might become necessary to instead feed a larger buffer that flushes more slowly, 
+	 // so as to minimize the overhead (both time and data) incurred by the HTTP transactions
+	// Thus the QUEUE is buffering data between successive iterations of firmwareLoop(), and 
+	// our "sendBuffer" will buffer up multiple QUEUEs before flushing on a time and/or data watermark.
+
+	// pop bytes from the device send queue (stop short of sendBuffer overflow)
+	while(!QUEUE_EMPTY(uint8_t, &device->sendQueue) && (pSendBuffer - sendBuffer) < sendBufferSize)
+	{
+        *pSendBuffer++ = QUEUE_POP(uint8_t, &device->sendQueue);
+    }
+
+	return;
+
+}
+
+#endif
```

Comment:
Agreed, I'll look at splitting this up. At minimum the server API stuff should be split off from the telit controller.